src/horust/supervisor/process_spawner.rs

```diff
@@ -169,14 +170,26 @@ fn redirect_output(
     Ok(())
 }
 
+/// Find program on PATH.
+///
+fn find_program_path(program_name: &String) -> std::result::Result<String, Errno> {
+    for dir in std::env::var("PATH").unwrap().split(":") {
+        let path = format!("{}/{}", dir, program_name);
+        if std::path::Path::new(&path).exists() {
+            return Ok(path);
+        }
+    }
+    Err(ENOENT)
+}
+
 /// Exec wrapper.
 ///
 /// # Safety
 ///
 /// Use only async-signal-safe, otherwise it might lock.
 #[inline]
 fn exec(
-    program_name: CString,
+    program_name: &String,
```

Comment:
Usually in rust you might want to use &str (for references to Strings) or String